### DISCUSSION TITLE
Why deep link got retriggered after page reload? How to clear the used deep link?

### DISCUSSION THREAD

**Discussion by HuakunShen** [tauri-apps/tauri#11591] [Q&A]
I am using deep link to do OAuth. It redirects to a auth callback page when triggered.
But when the page is reloaded (while I am on home page `"/"`), the old deep link event still gets triggered.
I am using `onOpenUrl()` in JS. 
Is there a way to clear the used deep links?

**Comment by FabianLars** [comment]
onOpenUrl calls getCurrent on registration. can't remember why but imo it's a mistake. I'll open a PR to revert this.

**Reply by FabianLars** [reply]
in the meantime you can register a standard tauri event listener and listen to `deep-link://new-url`

**Reply by HuakunShen** [reply]
Yeah, I checked `getCurrent` value, it's always the last deep links triggered.

What is `deep-link://new-url`? What is a deep-link://new-url used for?

**Reply by FabianLars** [reply]
this is the event the plugin will emit when a deep link is triggered == when your oauth flow redirects to the deep link.

for reference this is how onOpenUrl is implemented right now: https://github.com/tauri-apps/plugins-workspace/blob/v2/plugins/deep-link/guest-js/index.ts#L99
